R/utils_render_latex.R

```diff
@@ -237,30 +233,23 @@ create_body_component_l <- function(data) {
   body <- dt_body_get(data = data)
   boxh <- dt_boxhead_get(data = data)
 
-  list_of_summaries <- dt_summary_df_get(data = data)
-  summaries_present <- dt_summary_exists(data = data)
   groups_rows_df <- dt_groups_rows_get(data = data)
-  stub_components <- dt_stub_components(data = data)
-
-  # Obtain the `styles_tbl` (NOTE: this is currently unused)
-  styles_tbl <- dt_styles_get(data = data)
 
-  n_data_cols <- length(dt_boxhead_get_vars_default(data = data))
   n_rows <- nrow(body)
 
-  # Determine whether the stub is available
-  stub_available <- dt_stub_components_has_rowname(stub_components = stub_components)
+  # Get vector representation of stub layout
+  stub_layout <- get_stub_layout(data = data)
 
   # Obtain all of the visible (`"default"`), non-stub
   # column names for the table
   default_vars <- dt_boxhead_get_vars_default(data = data)
 
   # Split `body_content` by slices of rows in cases where there is
   # and isn't a stub present
-  if (stub_available) {
+  if (length(stub_layout) > 0) {
```

Comment:
Don't check for `length(stub_layout)>0 here; both branches can use get_body_component_cell_matrix